src/modules/m_sqloper.cpp

```diff
@@ -19,49 +20,110 @@
 
 #include "inspircd.h"
 #include "modules/sql.h"
-#include "modules/hash.h"
 
-class OperQuery : public SQL::Query
+class OpsGetQuery : public SQL::Query
 {
  public:
+	// This variable will store all the OPER blocks from the DB
+	std::vector<std::string>& my_blocks;
+	/** We want to store the username and password if this is called during an /OPER, as we're responsible for /OPER post-DB fetch
+	 *  Note: uid will be empty if this DB update was not called as a result of a user command (i.e. /REHASH)
+	 */
 	const std::string uid, username, password;
-	OperQuery(Module* me, const std::string& u, const std::string& un, const std::string& pw)
+	OpsGetQuery(Module* me, std::vector<std::string>& mb, const std::string& u, const std::string& un, const std::string& pw)
 		: SQL::Query(me)
+		, my_blocks(mb)
 		, uid(u)
 		, username(un)
 		, password(pw)
 	{
 	}
+	OpsGetQuery(Module* me, std::vector<std::string>& mb)
+		: SQL::Query(me)
+		, my_blocks(mb)
+	{
+	}
 
 	void OnResult(SQL::Result& res) CXX11_OVERRIDE
 	{
-		ServerInstance->Logs->Log(MODNAME, LOG_DEBUG, "result for %s", uid.c_str());
-		User* user = ServerInstance->FindNick(uid);
-		if (!user)
-			return;
+		ServerConfig::OperIndex& oper_blocks = ServerInstance->Config->oper_blocks;
+
+		// Remove our previous blocks from oper_blocks for a clean update
+		for (std::vector<std::string>::const_iterator i = my_blocks.begin(); i != my_blocks.end(); ++i)
+		{
+			oper_blocks.erase(*i);
+		}
+		my_blocks.clear();
 
-		// multiple rows may exist
 		SQL::Row row;
+		// Iterate through DB results to create oper blocks from sqloper rows
 		while (res.GetRow(row))
 		{
-			if (OperUser(user, row[0], row[1]))
-				return;
+			parameterlist cols;
```

Comment:
This is an inappropriate usage of a protocol interface typedef. Just use a std::vector<std::string>.